StyleCop.Analyzers/StyleCop.Analyzers/Helpers/DocumentationCommentExtensions.cs

```diff
@@ -0,0 +1,190 @@
+
+            int removeFromEnd;
+            if (IsXmlNewLine(lastSyntaxTokens[lastSyntaxTokens.Count - 1]))
+            {
+                removeFromEnd = 1;
+            }
+            else
+            {
+                if (!IsXmlWhitespace(lastSyntaxTokens[lastSyntaxTokens.Count - 1]))
+                    return summaryContent;
+
+                if (!IsXmlNewLine(lastSyntaxTokens[lastSyntaxTokens.Count - 2]))
+                    return summaryContent;
+
+                removeFromEnd = 2;
+            }
+
+            for (int i = 0; i < removeFromStart; i++)
+            {
+                firstSyntaxTokens = firstSyntaxTokens.RemoveAt(0);
+            }
+
+            if (firstSyntax == lastSyntax)
+            {
+                lastSyntaxTokens = firstSyntaxTokens;
+            }
+
+            for (int i = 0; i < removeFromEnd; i++)
+            {
+                lastSyntaxTokens = lastSyntaxTokens.RemoveAt(lastSyntaxTokens.Count - 1);
+            }
+
+            summaryContent = summaryContent.RemoveAt(summaryContent.Count - 1);
+            if (lastSyntaxTokens.Count != 0)
+                summaryContent = summaryContent.Add(lastSyntax.WithTextTokens(lastSyntaxTokens));
+
+            if (firstSyntax != lastSyntax)
+            {
+                summaryContent = summaryContent.RemoveAt(0);
+                if (firstSyntaxTokens.Count != 0)
+                    summaryContent = summaryContent.Insert(0, firstSyntax.WithTextTokens(firstSyntaxTokens));
+            }
+
+            if (summaryContent.Count > 0)
+            {
+                // Make sure to remove the leading trivia
+                summaryContent = summaryContent.Replace(summaryContent[0], summaryContent[0].WithLeadingTrivia());
+
+                // Remove leading spaces (between the <para> start tag and the start of the paragraph content)
+                XmlTextSyntax firstTextSyntax = summaryContent[0] as XmlTextSyntax;
+                if (firstTextSyntax != null && firstTextSyntax.TextTokens.Count > 0)
+                {
+                    SyntaxToken firstTextToken = firstTextSyntax.TextTokens[0];
+                    string firstTokenText = firstTextToken.Text;
+                    string trimmed = firstTokenText.TrimStart();
+                    if (trimmed != firstTokenText)
+                    {
+                        SyntaxToken newFirstToken = SyntaxFactory.Token(
+                            firstTextToken.LeadingTrivia,
+                            firstTextToken.Kind(),
+                            trimmed,
+                            firstTextToken.ValueText.TrimStart(),
+                            firstTextToken.TrailingTrivia);
+
+                        summaryContent = summaryContent.Replace(firstTextSyntax, firstTextSyntax.ReplaceToken(firstTextToken, newFirstToken));
+                    }
+                }
+            }
+
+            return summaryContent;
+        }
```

Comment:
@sharwell I can't really think of a better name as well. 
As far as I'm concerned there is no need to revisit the behavior. I would prefer documentation for the method that explains the additional behavior, so that its clear for other developers reading the code that the method does more than the name implies.